java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/htmlui/Browser.java

```diff
@@ -18,538 +18,167 @@
  */
 package org.netbeans.modules.java.lsp.server.htmlui;
 
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
 import java.io.Closeable;
-import java.io.File;
-import java.io.FileNotFoundException;
-import java.io.Flushable;
 import java.io.IOException;
-import java.io.InputStream;
 import java.io.InputStreamReader;
-import java.io.OutputStream;
 import java.io.Reader;
-import java.io.Writer;
-import java.net.URI;
-import java.net.URISyntaxException;
 import java.net.URL;
-import java.net.URLConnection;
 import java.net.URLDecoder;
-import java.nio.file.Files;
-import java.util.ArrayList;
+import java.util.Collections;
 import java.util.HashMap;
-import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
-import java.util.Queue;
 import java.util.Random;
+import java.util.Set;
 import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
 import java.util.function.Consumer;
-import java.util.function.Supplier;
 import java.util.logging.Level;
 import java.util.logging.Logger;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.CodeActionParams;
 import org.netbeans.html.boot.spi.Fn;
-import org.netbeans.html.boot.spi.Fn.Presenter;
 import org.netbeans.html.presenters.spi.ProtoPresenter;
 import org.netbeans.html.presenters.spi.ProtoPresenterBuilder;
+import org.netbeans.modules.java.lsp.server.protocol.CodeActionsProvider;
+import org.netbeans.modules.java.lsp.server.protocol.HtmlPageParams;
+import org.netbeans.modules.java.lsp.server.protocol.NbCodeLanguageClient;
+import org.netbeans.modules.parsing.api.ResultIterator;
+import org.netbeans.spi.htmlui.HTMLViewerSpi;
+import org.openide.util.Exceptions;
+import org.openide.util.lookup.ServiceProvider;
 
-/** Browser based {@link Presenter}. It starts local server and
- * launches browser that connects to it. Use {@link Browser.Config} to
- * configure the actual browser to be started.
- * <p>
- * To use this presenter specify following dependency:
- * <pre>
- * &lt;dependency&gt;
- *   &lt;groupId&gt;org.netbeans.html.browser&lt;/groupId&gt;
- *   &lt;artifactId&gt;browser&lt;/artifactId&gt;
- *   &lt;version&gt;<a target="blank" href="http://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22com.dukescript.presenters%22%20AND%20a%3A%22browser%22">1.x</a>&lt;/version&gt;
- * &lt;/dependency&gt;
- * </pre>
- */
-public final class Browser implements Fn.Presenter, Fn.KeepAlive, Flushable,
-Executor, Closeable {
-    static final Logger LOG = Logger.getLogger(Browser.class.getName());
-    private final Map<String,Command> SESSIONS = new HashMap<>();
+public class Browser implements Closeable {
```

Comment:
I suggest to rename this class. It is no longer about "browser".

Reply:
Renamed to `WebView`.